Core/CoreTests/Common/CommonUI/CoreWebView/Model/CoreWebViewStudioFeaturesInteractorTests.swift

```diff
@@ -199,10 +199,33 @@ class CoreWebViewStudioFeaturesInteractorTests: CoreTestCase {
 
         // When
         let actionUrl = "https://suhaibalabsi.instructure.com/media_attachments/546734/immersive_view?title=Hello%20World"
-        let action = MockNavigationAction(url: actionUrl, type: .linkActivated, targetFrame: MockFrameInfo(isMainFrame: false))
+        let action = MockNavigationActionRepresentable(url: actionUrl, type: .linkActivated, targetFrame: MockInfoFrameInfoRepresentable(isMainFrame: false))
         let immersiveUrl = interactor.urlForStudioImmersiveView(of: action)
 
         // Then
         XCTAssertEqual(immersiveUrl?.absoluteString, "https://suhaibalabsi.instructure.com/media_attachments/546734/immersive_view?title=Hello%20World&embedded=true")
     }
 }
+
+struct MockInfoFrameInfoRepresentable: FrameInfoRepresentable {
+    let isMainFrame: Bool
+}
+
+struct MockNavigationActionRepresentable: NavigationActionRepresentable {
```

Comment:
Please make both private
```suggestion
private struct MockInfoFrameInfoRepresentable: FrameInfoRepresentable {
    let isMainFrame: Bool
}

private struct MockNavigationActionRepresentable: NavigationActionRepresentable {
```